### ISSUE TITLE
cipher-aes-0.2.10 test suite failure

### ISSUE THREAD

**Issue by peti** [vincenthz/hs-cipher-aes#35]
Citing from http://hydra.cryp.to/build/910478/log/raw:

~~~~~~~~~~~
Linking dist/build/test-cipher-aes/test-cipher-aes ...
running tests
Running 1 test suites...
Test suite test-cipher-aes: RUNNING...
Test suite test-cipher-aes: FAIL
~~~~~~~~~~~

**Comment by snoyberg** [comment]
I just ran into this too, and it turned out to be a segfault:

```
  decrypt.encrypt==id:
    ECB: [OK, passed 100 tests]
    CBC: [OK, passed 100 tests]
    CFB: [OK, passed 100 tests]
    CFB8: [OK, passed 100 tests]
    CTR: [OK, passed 100 tests]
    OCB: [OK, passed 100 tests]
    CCM: [OK, passed 100 tests]
    EAX: [OK, passed 100 tests]
    CWC: [OK, passed 100 tests]
    GCM: [OK, passed 100 tests]
Segmentation fault (core dumped)
```

This occurs with the dependencies coming from lts-2.11, in case that helps.

**Comment by vincenthz** [comment]
Thanks. That was an annoying one to track down; It was related to some hasty compilation fixes on 32 bit that I've did last week. I've fixed the underlaying issue in memory (released as memory 0.7, and made a securemem 0.1.9 that put a lower bound at 0.7)

**Comment by snoyberg** [comment]
Thanks for the fix. Given severity of this, I'm releasing a new LTS version immediately that includes the new version of memory.
